ykrt/src/trace/swt/mod.rs

```diff
@@ -0,0 +1,133 @@
+//! Software tracer.
+
+use crate::frame::BitcodeSection;
+
+use super::{errors::InvalidTraceError, AOTTraceIterator, TraceAction, TraceRecorder};
+use std::sync::Once;
+use std::{cell::RefCell, collections::HashMap, error::Error, ffi::CString, sync::Arc};
+
+static FUNC_NAMES_INIT: Once = Once::new();
+
+#[derive(Debug, Eq, PartialEq)]
+struct TracingBlock {
+    function_index: usize,
+    block_index: usize,
+}
+
+thread_local! {
+    static BASIC_BLOCKS: RefCell<Vec<TracingBlock>> = RefCell::new(vec![]);
+    static FUNC_NAMES: RefCell<HashMap<usize, CString>> = RefCell::new(HashMap::new());
```

Comment:
What is `FUNC_NAMES`? Let's add (even brief) doc strings to these things so that dummies like me have some help :)

Reply:
Added 👉 b3cd5fd0b66ac9a0a8adfb0ab58aa0bbcbd4cedb